funsor/gaussian.py

```diff
@@ -635,12 +670,21 @@ def eager_add_gaussian_gaussian(op, lhs, rhs):
     # Align data.
     inputs = lhs.inputs.copy()
     inputs.update(rhs.inputs)
-    lhs_info_vec, lhs_precision = align_gaussian(inputs, lhs)
-    rhs_info_vec, rhs_precision = align_gaussian(inputs, rhs)
+    lhs_info_vec, lhs_precision = align_gaussian(inputs, lhs, try_keeping_block_form=True)
+    keep_block = isinstance(lhs_info_vec, BlockVector)
+    rhs_info_vec, rhs_precision = align_gaussian(inputs, rhs, try_keeping_block_form=keep_block)
+
+    if keep_block and not isinstance(rhs_info_vec, BlockVector):
```

Comment:
I think you'll also need to test whether the two block forms are compatible. E.g. the following two block forms are not compatible:
- `[0:1]`, `[1:3]`
- `[0:2]`, `[2:3]`